Common/TableProducer/centralityTable_new.cxx

```diff
@@ -0,0 +1,149 @@
+              LOGF(info, "Table %s disabled", table.c_str());
+            }
+          }
+        };
+        enable("V0M", estV0M);
+        enable("SPD", estSPD);
+        enable("CL0", estCL0);
+        enable("CL1", estCL1);
+      }
+    }
+    ccdb->setURL("http://alice-ccdb.cern.ch");
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    mRunNumber = 0;
+    mV0MCallibrationStored = false;
+    mhVtxAmpCorrV0A = nullptr;
+    mhVtxAmpCorrV0C = nullptr;
+    mhMultSelCalibV0M = nullptr;
+  }
+
+  void process(soa::Join<aod::Collisions, aod::Mults>::iterator const& collision, aod::BCsWithTimestamps const&, aod::Tracks const& tracks)
+  {
+    /* check the previous run number */
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (bc.runNumber() != mRunNumber) {
+      LOGF(debug, "timestamp=%llu", bc.timestamp());
+      TList* callst = ccdb->getForTimeStamp<TList>("Users/v/victor/Centrality", bc.timestamp());
+
+      if (callst != nullptr) {
+        auto zvtxcalib = [callst](const char* tpname, const char* hname) {
+          TProfile* tp = (TProfile*)callst->FindObject(tpname);
+          TH1* h = new TH1F(TString::Format("%s", hname), TString::Format("%s;#it{z}_{vtx}", hname),
+                            tp->GetNbinsX(), tp->GetXaxis()->GetBinLowEdge(1), tp->GetXaxis()->GetBinUpEdge(tp->GetNbinsX()));
+          TF1 one = TF1("one", "1", tp->GetXaxis()->GetBinLowEdge(1), tp->GetXaxis()->GetBinUpEdge(tp->GetNbinsX()));
+          h->Reset();
+          h->Add(&one, tp->GetMean(2));
+          h->Divide(tp);
+          return h;
+        };
+        auto multcalib = [callst](const char* ccdbhname, const char* hname) {
+          TH1* h = (TH1*)callst->FindObject(ccdbhname)->Clone(hname);
+          return h;
+        };
+        if (estV0M == 1) {
+          mhVtxAmpCorrV0A = zvtxcalib("hVtx_fAmplitude_V0A", "zvtxCalibV0A");
+          mhVtxAmpCorrV0C = zvtxcalib("hVtx_fAmplitude_V0C", "zvtxCalibV0C");
+          mhMultSelCalibV0M = multcalib("hMultSelCalib_V0M", "MultSelCalibV0M");
+          if ((mhVtxAmpCorrV0A != nullptr) and (mhVtxAmpCorrV0C != nullptr) and (mhMultSelCalibV0M != nullptr)) {
+            mV0MCallibrationStored = true;
+          }
+        }
+        if (estSPD == 1) {
+          LOGF(fatal, "Calibraltion information estimated from SPD tracklets still not available");
+        }
+        if (estCL0 == 1) {
+          LOGF(fatal, "Calibraltion information estimated from CL0 still not available");
+        }
+        if (estCL1 == 1) {
+          LOGF(fatal, "Calibraltion information estimated from CL1 still not available");
+        }
+        if (mV0MCallibrationStored) {
+          mRunNumber = bc.runNumber();
+        }
+      } else {
+        /* we dont change the run number to keep trying */
+        mV0MCallibrationStored = false;
+        LOGF(error, "Centrality calibration is not available in CCDB for run=%d at timestamp=%llu", bc.runNumber(), bc.timestamp());
```

Comment:
So if it is not found, we make one request per collision? This is not good I think because it makes a high load on the CCDB. I would not retry for this run then.

Reply:
Shouldn't the error be fatal?

Reply:
My logic was
a) if there is a glitch in the CCDB we don't discard the whole run, let's keep trying
b) if a run does not have calibration information perhaps others do, don't kill the workflow

But you are right a) will originate high load towards CCDB, and b) will make no sense if the CCDB is down for any reason.

I can set counters for both cases and act accordingly
- one counter for the number of tries on a concrete run number before we give up trying for that run and set all centralities out of range for that run
- one counter for the number of runs without CCDB access before we kill the workflow

or just act without any counters at all

Reply:
I don't think we need failsafe at many levels (CCDB will already try to get the file from all the replicas). We also resubmit the job on failure.
If we ask centrality for a run which is not available, it is better to bail out. So I would go for 
- no counter
- Fatal, right away

Reply:
Fatal right away if the centrality calibration is not available or if it is corrupted (not the whole set of histograms are present)